library/iterator.lisp

```diff
@@ -517,6 +520,33 @@ Return `None` if ITER is empty."
     "Return the most-negative element of ITER, or `None` if ITER is empty."
     (optimize! < iter))
 
+  (declare optimizing! ((:b -> :b -> Boolean) ->
```

Comment:
I think we need to adjust the name. Maybe `optimize-by!`?

I'm not opposed to "optimizing" and the like, but this will be confused with "optimize", and also it's like "argmin"/"argmax", not the max of
f.